### PR TITLE
Fix grad accum arbitrary value

### PR DESCRIPTION
**Pull request by IlyasMoutawwakil**
# What does this PR do?

The test for the "brokenness" of gradient accumulation loss uses an arbitrary value to check if not passing "num_items_in_batch" results in a loss different from normal training loss. This value got smaller after I made some changes to the test (mostly trying to make it more understandable).
See changes in https://github.com/huggingface/transformers/pull/36424/files#diff-ac9b69e204d41f7ddf23e8bd5b53a2a0ffa64c4784101abeef3b5d4c14342656R823-R905

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Review by SunMarc — APPROVED** [review]
Works for me !

**Review by Cyrilvallez — APPROVED** [review]
Alright, LGTM! Thanks a lot!

**Review by shethaadit — APPROVED** [review]
